include/deal.II/multigrid/mg_tools.h

```diff
@@ -198,6 +198,18 @@ namespace MGTools
                       const ComponentMask               &component_mask = ComponentMask());
 
   /**
+  * The same function as above, but return an IndexSet rather than a
+  * std::set<unsigned int> on each level and use a std::set of
+  * boundary_indicators as input
+  */
+  template <int dim, int spacedim>
+  void
+  make_boundary_list (const DoFHandler<dim,spacedim>      &mg_dof,
+                      const std::set<types::boundary_id> &boundary_indicators,
+                      std::vector<IndexSet>                 &boundary_indices,
+                      const ComponentMask               &component_mask = ComponentMask());
```

Comment:
We've started moving away from returning objects through reference arguments, in particular if they are followed by other input arguments (possibly defaulted).

Both `std::vector` and `IndexSet` now have move constructors and operators. How about just doing as the documentation suggests and instead *returning* `boundary_indices`?